internal/sqs/sqs.go

```diff
@@ -0,0 +1,87 @@
+package ecrconsumer
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+
+	"github.com/Wattpad/sqsconsumer"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+)
+
+type SQSConfig struct {
+	Name   string
+	Region string
+	Svc    *sqs.SQS
+}
+
+func NewSQSConfig(name string, region string) *SQSConfig {
+	conf := &aws.Config{
+		Region: &region,
+	}
+	svc := sqs.New(session.New(conf))
+
+	return &SQSConfig{
+		Name:   name,
+		Region: region,
+		Svc:    svc,
+	}
+}
+
+func SQSObjectForQueue(name string, svc *sqs.SQS) (*sqsconsumer.SQSService, error) {
+	return nil, nil
+}
+
+func (s *SQSConfig) NewSQSConsumer() (*sqsconsumer.Consumer, error) {
+	// Create SQS service
+	service, err := SQSObjectForQueue(s.Name, s.Svc)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create and return SQS consumer
+	consumer := sqsconsumer.NewConsumer(service, processMessage)
+	consumer.SetLogger(log.Printf)
+	return consumer, nil
+}
+
+// Runs a preconfigured Consumer
+func RunConsumer(c *sqsconsumer.Consumer) {
+
+	numFetchers := 1
+
+	// set up a context which will gracefully cancel the worker on interrupt
+	fetchCtx, cancelFetch := context.WithCancel(context.Background())
+	term := make(chan os.Signal, 1)
+	signal.Notify(term, os.Interrupt, os.Kill)
```

Comment:
`signal.Notify(term, syscall.SIGINT, syscall.SIGTERM)`

Reply:
That worked. Thank you!